tools/profileBuilder/model/transforms.go

```diff
@@ -35,155 +35,127 @@ import (
 	"path/filepath"
 	"regexp"
 	"strings"
+	"sync"
 	"time"
 
-	"github.com/marstr/collection"
-	goalias "github.com/marstr/goalias/model"
 	"golang.org/x/tools/imports"
 )
 
-type Alias struct {
-	*goalias.AliasPackage
-	TargetPath string
+// ListDefinition represents a JSON file that contains a list of packages to include
+type ListDefinition struct {
+	Include []string `json:"include"`
 }
 
-const armPathModifier = "mgmt"
+const (
+	armPathModifier = "mgmt"
+	aliasFileName   = "models.go"
+)
 
-var packageName = regexp.MustCompile(`services[/\\](?P<provider>[\w\-\.\d_\\/]+)[/\\](?:(?P<arm>` + armPathModifier + `)[/\\])?(?P<version>v?\d{4}-\d{2}-\d{2}[\w\d\.\-]*|v?\d+\.\d+[\.\d\w\-]*)[/\\](?P<group>[/\\\w\d\-\._]+)`)
+var packageName = regexp.MustCompile(`services[/\\](?P<provider>[\w\-\.\d_\\/]+)[/\\](?:(?P<arm>` + armPathModifier + `)[/\\])?(?P<version>v?\d{4}-\d{2}-\d{2}[\w\d\.\-]*|v?\d+[\.\d+\.\d\w\-]*)[/\\](?P<group>[/\\\w\d\-\._]+)`)
 
 // BuildProfile takes a list of packages and creates a profile
-func BuildProfile(packageList collection.Enumerable, name, outputLocation string, outputLog, errLog *log.Logger) {
-	var packages collection.Enumerator
-
-	// Find the names of all of the packages for inclusion in this profile.
-	packages = packageList.Enumerate(nil).Select(func(x interface{}) interface{} {
-		if cast, ok := x.(string); ok {
-			return cast
-		}
-		return nil
-	})
-
-	// Parse the packages that were selected for inclusion in this profile.
-	packages = packages.SelectMany(func(x interface{}) collection.Enumerator {
-		results := make(chan interface{})
-
-		go func() {
-			defer close(results)
-
-			cast, ok := x.(string)
-			if !ok {
-				return
+func BuildProfile(packageList ListDefinition, name, outputLocation string, outputLog, errLog *log.Logger) {
+	wg := &sync.WaitGroup{}
+	wg.Add(len(packageList.Include))
+	for _, pkgDir := range packageList.Include {
+		if !filepath.IsAbs(pkgDir) {
+			abs, err := filepath.Abs(pkgDir)
+			if err != nil {
+				errLog.Fatalf("failed to convert to absolute path: %v", err)
 			}
-			files := token.NewFileSet()
-			parsed, err := parser.ParseDir(files, cast, nil, 0)
+			pkgDir = abs
+		}
+		go func(pd string) {
+			fs := token.NewFileSet()
+			packages, err := parser.ParseDir(fs, pd, func(f os.FileInfo) bool {
```

Comment:
hmm can we extract the goroutine as a seperate function ?

Reply:
We can however I did it this way since nobody else needs to call this function (it's an implementation detail of `BuildProfile()`).  If you feel strongly about it I will move it.

Reply:
oh was mostly just for better readability